submissions/bmukha/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,73 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/bmukha/a-tiny-JS-world
+   Web app: https://bmukha.github.io/a-tiny-JS-world/
+   */
+
+class Character {
+  constructor(specie, name, gender, saying, friends, legs) {
+    this.specie = specie;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.saying = saying;
+    this.friends = friends;
+  }
+
+  props() {
+    return ['specie', 'name', 'gender', 'legs', 'hands', 'saying', 'friends']
+      .map((prop) =>
+        Array.isArray(this[prop]) ? this[prop].join(', ') : this[prop]
+      )
+      .filter((value) => Boolean(value))
```

Comment:
I'm a bit confused about this one. In the previous version of this task, I deliberately rewrote that piece this way after your suggestions. And it really simplified my code because I can get rid of all the falsy values with one filter (because I convert possible empty arrays (still truthy values) into empty strings (falsy values) beforehand. Here are the links to our discussion:
[Your suggestion](https://github.com/kottans/frontend-2022-homeworks/pull/193#discussion_r948902740)
[My reply](https://github.com/kottans/frontend-2022-homeworks/pull/193#discussion_r949157543)
So, unless I'm missing something, I guess it is better to stick to this realisation. What do you think?

Reply:
Oh, I see. Makes sense.
I was thinking of rather `null` or `undefined` to denote `no friends` situation.
Empty array is a valid approach. Keep this fragment as it is.